lib/core/src/rcMisc.cpp

```diff
@@ -4729,3 +4757,43 @@ myRead( int sock, void *buf, int len,
     }
     return len - toRead;
 }
+
+auto resolve_hostname(const std::string_view _hostname,
+                      const nlohmann::json& _hosts_config,
+                      hostname_resolution_scheme _scheme) -> std::optional<std::string>
+{
+    using json = nlohmann::json;
+
+    const auto host_entries = _hosts_config.at("host_entries");
+    const auto end = std::end(host_entries);
+
+    // Find the addresses object that contains the address, "_hostname".
+    const auto iter = std::find_if(std::begin(host_entries), end, [_hostname](const json& _entry) {
+        auto&& addresses = _entry.at("addresses");
+        const auto end = std::end(addresses);
+        return end != std::find_if(std::begin(addresses), end, [_hostname](const json& _address) {
+            return _hostname == _address.at("address").get<std::string>();
+        });
+    });
+
+    if (iter == end) {
+        return std::nullopt;
+    }
+
+    switch (_scheme) {
+        case hostname_resolution_scheme::match_preferred:
+            return iter->at("addresses").front().at("address").get<std::string>();
+
+        case hostname_resolution_scheme::match_longest:
+            std::string longest;
+
+            for (auto&& address : iter->at("addresses")) {
+                if (auto tmp = address.at("address").get<std::string>(); tmp.size() > longest.size()) {
+                    longest = std::move(tmp);
+                }
+            }
+
+            return longest;
```

Comment:
Should we handle the default case? I realize it's an enum on which we are switching here, just makes me itch.

Also, my C brain did a double-take when I saw the declaration without curly braces :)

Reply:
add a comment about the `default` keyword not being included.

Reply:
On second thought, handling the default case is the better option. If someone casts a random integer to a `hostname_resolution_scheme`, who knows what would happen. At least, the default case would handle that safely.